File: 0822-unique-morse-code-words/0822-unique-morse-code-words.go
import "fmt"

func uniqueMorseRepresentations(words []string) int {
    TABLE := []string{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."}
    MORSE_MAP := make(map[string]string)
    answer_set := make(map[string]bool)
    var answer int 
    for i:=0;i<len(TABLE);i++{
         s := fmt.Sprintf("%c", 97+i)
         MORSE_MAP[s] = TABLE[i]
    }
    for i:=0;i<len(words);i++ {
        morse_code := ""
        for j:=0;j<len(words[i]);j++{
            morse_code += MORSE_MAP[string(words[i][j])]
        }
        answer_set[morse_code] = true
    }
    for k := range answer_set {
        if answer_set[k] {
            answer ++
        }
    }
    return answer
}

File: 0822-unique-morse-code-words/0822-unique-morse-code-words.py
class Solution:
    TABLE = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
    def __init__(self):
        self.MORSE_CODE_TABLE = {}
        self.tableToAlphaMap()

    def tableToAlphaMap(self):
        for i in range(len(self.TABLE)):
            self.MORSE_CODE_TABLE[chr(97+i)] = self.TABLE[i]

    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        answer_set = set()
        for word in words:
            morse_code = ''
            for c in word:
                morse_code += self.MORSE_CODE_TABLE[c]
            answer_set.add(morse_code)
        return len(answer_set)